### PR TITLE
Handle user input for vendor and purpose IDs

### PR DESCRIPTION
**Pull request by JoshuaRogan**

### CODE REVIEW

File: src/ConsentManagementProvider.js [hunk 1]
```diff
@@ -336,11 +336,11 @@ class ConsentManagementProvider {
             consents = {
                 metadata: this.vendorConsent.getMetadataString(),
                 purposeConsents: toAllowedMap(
-                    createIdArray(1, MAX_STANDARD_PURPOSE_ID),
+                    createIdArray(1, MAX_STANDARD_PURPOSE_ID), // get from user
```
Comment (dmnsobczak) [inline src/ConsentManagementProvider.js]: 👍

File: src/ConsentManagementProvider.js [hunk 2]
```diff
@@ -336,11 +336,11 @@ class ConsentManagementProvider {
...
                     (id) => this.vendorConsent.isPurposeAllowed(id)
                 ),
                 vendorConsents: toAllowedMap(
-                    vendorIds,
+                    vendorIds, // get from user
```
Comment (dmnsobczak) [inline src/ConsentManagementProvider.js]: 👍

File: src/OptInManager.js [hunk 3]
```diff
@@ -52,6 +61,10 @@ class OptInManager {
 
         Cookies.set(name, value, attributes);
         Cookies.set(VERSION_COOKIE_NAME, CMP_VERSION, attributes);
+
+        // set a new cookie with the vendor list and purpose list as JSON
+        const cookieValue = {vendorIds: this.vendorIds, purposeIds: this.purposeIds};
+        Cookies.set(VENDOR_PURPOSE_LIST_COOKIE_NAME, JSON.stringify(cookieValue), attributes);
```
Comment (dmnsobczak) [inline src/OptInManager.js]: Do we really need a new cookie for that? Those values are stored inside the consent string:
https://github.com/Wikia/tracking-opt-in/blob/master/src/ConsentManagementProvider.js#L41

File: src/ConsentManagementProvider.js (hunk 1 above)
Comment (JoshuaRogan) [reply src/ConsentManagementProvider.js]: Hmm it looks like this PR in the current state properly saves the consent cookie. What is this part of the code for?

Comment (dmnsobczak) [reply src/ConsentManagementProvider.js]: Ook, I got it now. Try opening the console on Communities and type:
`window.__cmp('getVendorConsents', null, (consentData) => console.log(consentData));`
You will call cmp wrapper directly and execute `getVendorConsents` function from L316.

**Pull request merged**